File: app/src/main/java/com/example/sekkawa7da/ui/MainPage.java
package com.example.sekkawa7da.ui;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.coordinatorlayout.widget.CoordinatorLayout;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;
import androidx.viewpager.widget.ViewPager;

import android.content.Context;
import android.graphics.PorterDuff;
import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.util.AttributeSet;
import android.util.Log;
import android.util.Xml;
import android.view.Gravity;
import android.view.ViewGroup;
import android.widget.RelativeLayout;
import android.widget.Toast;

import com.example.sekkawa7da.R;
import com.example.sekkawa7da.MainPageNavigationFragments.HomeFragment;
import com.example.sekkawa7da.MainPageNavigationFragments.MoreFragment;
import com.example.sekkawa7da.MainPageNavigationFragments.NotificationsFragment;
import com.google.android.material.tabs.TabLayout;
import com.nex3z.notificationbadge.NotificationBadge;

import org.xmlpull.v1.XmlPullParser;

import java.util.ArrayList;
import java.util.List;

public class MainPage extends AppCompatActivity {

    RelativeLayout relativeLayout;

    private Toolbar toolbar;
    private ViewPager viewPager;
    private TabLayout tabLayout;

    private HomeFragment homeFragment;
    private NotificationsFragment notificationsFragment;
    private MoreFragment moreFragment;

    //private NotificationBadge notifyBadge;
    private int count=0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main_page);

        toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        viewPager = findViewById(R.id.view_pager);
        tabLayout = findViewById(R.id.tab_layout);

        homeFragment = new HomeFragment();
        notificationsFragment = new NotificationsFragment();
        moreFragment = new MoreFragment();

        tabLayout.setupWithViewPager(viewPager);

        ViewPagerAdapter viewPagerAdapter = new ViewPagerAdapter(getSupportFragmentManager() , 0);
        viewPagerAdapter.addFragment(homeFragment , "Home");
        viewPagerAdapter.addFragment(notificationsFragment , "Notifications");
        viewPagerAdapter.addFragment(moreFragment , "More");
        viewPager.setAdapter(viewPagerAdapter);

        //To Add Icons
        tabLayout.getTabAt(0).setIcon(R.drawable.ic_home_24dp);
        tabLayout.getTabAt(0).getIcon().setColorFilter(getResources().getColor(R.color.colorBlue), PorterDuff.Mode.SRC_IN);
        tabLayout.getTabAt(1).setIcon(R.drawable.ic_notifications_none_black_24dp);
        tabLayout.getTabAt(2).setIcon(R.drawable.ic_menu_black_24dp);

        //Notification Badge
        relativeLayout = findViewById(R.id.tab_lay_relative);
        NotificationBadge notifyBadge = new NotificationBadge(this, null);  //Create notification badge
        RelativeLayout.LayoutParams params = new RelativeLayout.LayoutParams(60,60); //set attributes for this badge
        params.leftMargin = 540;
        relativeLayout.addView(notifyBadge, params);
        notifyBadge.setNumber(++count); //number in this badge

        tabLayout.addOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
            @Override
            public void onTabSelected(TabLayout.Tab tab) {
                switch (tab.getPosition()){
                    case 0:
                        tabLayout.getTabAt(0).getIcon().setColorFilter(getResources().getColor(R.color.colorBlue), PorterDuff.Mode.SRC_IN);
                        break;
                    case 1:
                        tabLayout.getTabAt(1).setIcon(R.drawable.ic_selected_notifications_24dp);
                        notifyBadge.clear();
                        break;
                    case 2:
                        tabLayout.getTabAt(2).getIcon().setColorFilter(getResources().getColor(R.color.colorBlue), PorterDuff.Mode.SRC_IN);
                        break;
                    default:
                        tabLayout.getTabAt(0).getIcon().setColorFilter(getResources().getColor(R.color.colorGrey), PorterDuff.Mode.SRC_IN);
                        tabLayout.getTabAt(1).setIcon(R.drawable.ic_notifications_none_black_24dp);
                        tabLayout.getTabAt(2).getIcon().setColorFilter(getResources().getColor(R.color.colorGrey), PorterDuff.Mode.SRC_IN);
                        break;
                }
            }
            @Override
            public void onTabUnselected(TabLayout.Tab tab) {
                switch (tab.getPosition()) {
                    case 0:
                        tabLayout.getTabAt(0).getIcon().setColorFilter(getResources().getColor(R.color.colorGrey), PorterDuff.Mode.SRC_IN);
                        break;
                    case 1:
                        tabLayout.getTabAt(1).setIcon(R.drawable.ic_notifications_none_black_24dp);
                        break;
                    case 2:
                        tabLayout.getTabAt(2).getIcon().setColorFilter(getResources().getColor(R.color.colorGrey), PorterDuff.Mode.SRC_IN);
                        break;
                    default:
                        tabLayout.getTabAt(0).getIcon().setColorFilter(getResources().getColor(R.color.colorGrey), PorterDuff.Mode.SRC_IN);
                        tabLayout.getTabAt(1).setIcon(R.drawable.ic_notifications_none_black_24dp);
                        tabLayout.getTabAt(2).getIcon().setColorFilter(getResources().getColor(R.color.colorGrey), PorterDuff.Mode.SRC_IN);
                        break;
                }
            }
            @Override
            public void onTabReselected(TabLayout.Tab tab) {

            }
        });

        //BadgeDrawable badgeDrawable;

    }

    private class ViewPagerAdapter extends FragmentPagerAdapter {

        private List<Fragment> fragments = new ArrayList<>();
        private List<String> fragmentTitles = new ArrayList<>();

        public ViewPagerAdapter(@NonNull FragmentManager fm, int behavior) {
            super(fm, behavior);
        }

        public void addFragment(Fragment fragment , String title){
            fragments.add(fragment);
            fragmentTitles.add(title);
        }

        @NonNull
        @Override
        public Fragment getItem(int position) {
            return fragments.get(position);
        }

        @Override
        public int getCount() {
            return fragments.size();
        }

        @Nullable
        @Override
        public CharSequence getPageTitle(int position) {
            return fragmentTitles.get(position);
        }
    }

}

File: app/src/main/java/com/mahmoudrefaie/sekkawahda/ui/MainPage/MainPage.kt
package com.mahmoudrefaie.sekkawahda.ui.MainPage

import android.content.Intent
import android.graphics.PorterDuff
import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import androidx.viewpager.widget.ViewPager
import com.google.android.material.tabs.TabLayout
import com.google.android.material.tabs.TabLayout.OnTabSelectedListener
import com.mahmoudrefaie.sekkawahda.R
import com.mahmoudrefaie.sekkawahda.ui.Login.Login
import com.mahmoudrefaie.sekkawahda.ui.MainPage.NavigationFragments.Home.HomeFragment
import com.mahmoudrefaie.sekkawahda.ui.MainPage.NavigationFragments.MoreFragment
import com.mahmoudrefaie.sekkawahda.ui.MainPage.NavigationFragments.Notifications.NotificationAdapter
import com.mahmoudrefaie.sekkawahda.ui.MainPage.NavigationFragments.Notifications.NotificationsFragment
import de.hdodenhof.circleimageview.CircleImageView
import java.util.*

class MainPage : AppCompatActivity() {
    private var searchIcon:CircleImageView? = null
    private var toolbar: Toolbar? = null
    private var viewPager: ViewPager? = null
    private var tabLayout: TabLayout? = null
    private var homeFragment: HomeFragment? = null
    private var notificationsFragment: NotificationsFragment? = null
    private var moreFragment: MoreFragment? = null

    //private NotificationBadge notifyBadge;
    private var count = 0
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main_page)
        toolbar = findViewById(R.id.toolbar)
        toolbar?.setTitle("Sekka Wahda")
        setSupportActionBar(toolbar)
        viewPager = findViewById(R.id.view_pager)
        tabLayout = findViewById(R.id.tab_layout)
        homeFragment = HomeFragment()
        notificationsFragment = NotificationsFragment()
        moreFragment = MoreFragment()
        tabLayout?.setupWithViewPager(viewPager)
        val viewPagerAdapter = ViewPagerAdapter(supportFragmentManager, 0)
        viewPagerAdapter.addFragment(homeFragment!!, "Home")
        viewPagerAdapter.addFragment(notificationsFragment!!, "Notifications")
        viewPagerAdapter.addFragment(moreFragment!!, "More")
        viewPager?.setAdapter(viewPagerAdapter)

        searchIcon = findViewById(R.id.search)
        searchIcon?.setOnClickListener(View.OnClickListener {
            var intent = Intent(this,Search::class.java)
            startActivity(intent)
        })

        //To Add Icons
        tabLayout?.getTabAt(0)!!.setIcon(R.drawable.ic_home_24dp)
        tabLayout?.getTabAt(0)!!.icon!!.setColorFilter(resources.getColor(R.color.proj_sub_color), PorterDuff.Mode.SRC_IN)
        tabLayout?.getTabAt(1)!!.setIcon(R.drawable.ic_notifications_none_black_24dp)
        tabLayout?.getTabAt(2)!!.setIcon(R.drawable.ic_menu_black_24dp)

        //Notification Badge
        val notificationBadgeDrawable = tabLayout?.getTabAt(1)!!.orCreateBadge
        notificationBadgeDrawable.isVisible = true
        tabLayout?.addOnTabSelectedListener(object : OnTabSelectedListener {
            override fun onTabSelected(tab: TabLayout.Tab) {
                when (tab.position) {
                    0 -> tabLayout?.getTabAt(0)!!.icon!!.setColorFilter(resources.getColor(R.color.proj_sub_color), PorterDuff.Mode.SRC_IN)
                    1 -> {
                        tabLayout?.getTabAt(1)!!.setIcon(R.drawable.ic_selected_notifications_24dp)
                        //notifyBadge.clear();
                        notificationBadgeDrawable.isVisible = false
                    }
                    2 -> tabLayout?.getTabAt(2)!!.icon!!.setColorFilter(resources.getColor(R.color.proj_sub_color), PorterDuff.Mode.SRC_IN)
                    else -> {
                        tabLayout?.getTabAt(0)!!.icon!!.setColorFilter(resources.getColor(R.color.colorGrey), PorterDuff.Mode.SRC_IN)
                        tabLayout?.getTabAt(1)!!.setIcon(R.drawable.ic_notifications_none_black_24dp)
                        tabLayout?.getTabAt(2)!!.icon!!.setColorFilter(resources.getColor(R.color.colorGrey), PorterDuff.Mode.SRC_IN)
                    }
                }
            }

            override fun onTabUnselected(tab: TabLayout.Tab) {
                when (tab.position) {
                    0 -> tabLayout?.getTabAt(0)!!.icon!!.setColorFilter(resources.getColor(R.color.colorGrey), PorterDuff.Mode.SRC_IN)
                    1 -> tabLayout?.getTabAt(1)!!.setIcon(R.drawable.ic_notifications_none_black_24dp)
                    2 -> tabLayout?.getTabAt(2)!!.icon!!.setColorFilter(resources.getColor(R.color.colorGrey), PorterDuff.Mode.SRC_IN)
                    else -> {
                        tabLayout?.getTabAt(0)!!.icon!!.setColorFilter(resources.getColor(R.color.colorGrey), PorterDuff.Mode.SRC_IN)
                        tabLayout?.getTabAt(1)!!.setIcon(R.drawable.ic_notifications_none_black_24dp)
                        tabLayout?.getTabAt(2)!!.icon!!.setColorFilter(resources.getColor(R.color.colorGrey), PorterDuff.Mode.SRC_IN)
                    }
                }
            }

            override fun onTabReselected(tab: TabLayout.Tab) {}
        })
    }

    private inner class ViewPagerAdapter(fm: FragmentManager, behavior: Int) : FragmentPagerAdapter(fm, behavior) {
        private val fragments: MutableList<Fragment> = ArrayList()
        private val fragmentTitles: MutableList<String> = ArrayList()
        fun addFragment(fragment: Fragment, title: String) {
            fragments.add(fragment)
            fragmentTitles.add(title)
        }

        override fun getItem(position: Int): Fragment {
            return fragments[position]
        }

        override fun getCount(): Int {
            return fragments.size
        }

        override fun getPageTitle(position: Int): CharSequence? {
            return fragmentTitles[position]
        }
    }

    private fun updateUI() {
        Toast.makeText(this, "You should login", Toast.LENGTH_LONG).show()
        val intent = Intent(this, Login::class.java)
        startActivity(intent)
        finish()
    }

}